### PR TITLE
Crate docs for `apis` module

### LINKED ISSUE #263: crate docs: for `servers::apis`mod (Tracker API)
Parent issue: https://github.com/torrust/torrust-tracker/issues/252

### PR DESCRIPTION
**Pull request by josecelano**
Documentation for the `crate::servers::apis` module.

### CODE REVIEW

File: src/tracker/mod.rs [hunk 1]
```diff
@@ -412,7 +413,7 @@
 //!
 //! For example, the HTTP tracker would send an event like the following when it handles an `announce` request received from a peer using IP version 4.
 //!
-//! ```rust,ignore
+//! ```text
```
Comment (da2ce7) [inline src/tracker/mod.rs:416]: Why the changing from `rust,ignore` to `text`?

Comment (josecelano) [reply src/tracker/mod.rs:416]: I do not want to compile that piece of code, and I do not want to have an "ignore" test in the output like this:

```
& cargo test --doc
test src/tracker/mod.rs - tracker (line 83) - compile ... ignore
```

Sometimes the necessary setup to compile is too verbose for the documentation. For example:

```text
let announce_data = tracker.announce(&info_hash, &mut peer, &peer_ip).await;
```

That line would require building a tracker, a peer, etc.

We have the tests if you want to check a fully operational example; maybe I could add a link to the tests in these cases. 

I can also implement the full working example and [hide the lines that are not interesting for the documentation](https://rust-lang.github.io/mdBook/format/mdbook.html#hiding-code-lines).

Comment (josecelano) [reply src/tracker/mod.rs:416]: The link to how to hide code in docs is wrong. I've read somewhere else that you can do that, but I do not find it.

Comment (josecelano) [reply src/tracker/mod.rs:416]: The link was not wrong it's only that you need to prepend the hashtag to all the lines you want to hide:

```text
//! ```rust,no_run
//! # use torrust_tracker::tracker::peer;
//! # use torrust_tracker::shared::bit_torrent::info_hash::InfoHash;
//! # // ...
//! 
//! let announce_data = tracker.announce(&info_hash, &mut peer, &peer_ip).await;
//! ```
```

That would show only:

```rust

let announce_data = tracker.announce(&info_hash, &mut peer, &peer_ip).await;
```

**Comment by josecelano** [comment]
Hey, @da2ce7 @WarmBeer, I'm not going to include in the API docs information about how to set up Nginx or Apache as reverse proxies, with or without SSL. I think we can move that information to blog posts (tutorials) on the torrust.com site.

**Comment by josecelano** [comment]
ACK 81c4d7072103144e5bd55bbf4147b47190e0951e

**Comment by josecelano** [comment]
Hi @da2ce7 @WarmBeer, this is ready to review.

**Pull request merged**